src/Reactor.cc

```diff
@@ -386,6 +386,27 @@ vector<string> Reactor::getHwConfigs(string_view type)
 	return result;
 }
 
+vector<string> Reactor::getSetups()
+{
+	vector<string> result;
+	std::string_view extension = Reactor::SETUP_EXTENSION;
+	for (auto context = userDataFileContext(SETUP_DIR);
+	     const auto& path : context.getPaths()) {
+		foreach_file(path, [&](const std::string& /*fullName*/, std::string_view name) {
+			if (name.ends_with(extension)) {
+				name.remove_suffix(extension.size());
+				result.emplace_back(std::string(name));
```

Comment:
Drop the explicit `std::string` construction when calling `emplace_back`.

`emplace_back` accepts any type(s) of parameter(s) and uses those to construct the element in-place in the `vector`.
* Currently you pass a (manually constructed) `string` and then `emplace_back()` will use the `string(string&&)` move-constructor. In other words: we create a temporary string and then move it into the vector.
* But if you instead call `emplace_back(name)` (drop `std::string()`, so pass a `string_view` parameter), then `emplace_back()` will directly construct the string in-place, without an intermediate temporary.